src/dotnet/MLSearch.Service/Indexing/ChatIndexTrigger.cs

```diff
@@ -26,12 +26,11 @@ public virtual async Task OnCancelCommand(MLSearch_CancelChatIndexing e, Cancell
     // [EventHandler]
     public virtual async Task OnTextEntryChangedEvent(TextEntryChangedEvent eventCommand, CancellationToken cancellationToken)
     {
-        // TODO: The question is should we index search chats as well? (Probably yes)
-        var peerChatId = eventCommand.Entry.ChatId.PeerChatId;
-        if (peerChatId.IsNone || !peerChatId.HasUser(Constants.User.MLSearchBot.UserId)) {
-            // A normal conversation.
-            var e = new MLSearch_TriggerChatIndexing(eventCommand.Entry.ChatId);
-            await commander.Call(e, cancellationToken).ConfigureAwait(false);
-        }
+        bool shouldBeIndexed = await chatIndexFilter.ChatShouldBeIndexed(eventCommand.Entry.ChatId, cancellationToken).ConfigureAwait(false);
+        if (!shouldBeIndexed)
+            return;
```

Comment:
Here we have the check before sending command, and we have another one on its execution.

Reply:
Same as above